cli/azd/cmd/infra_create.go

```diff
@@ -272,25 +276,78 @@ func (ica *infraCreateAction) Run(ctx context.Context, cmd *cobra.Command, args
 	return nil
 }
 
-func reportDeploymentStatusInteractive(ctx context.Context, azCli tools.AzCli, env environment.Environment, showProgress func(string)) {
+func reportDeploymentStatusInteractive(ctx context.Context, azCli tools.AzCli, env environment.Environment, spinnerUpdater *spin.SpinnerUpdater, loggedResourcesPtr *map[string]bool) {
 	resourceManager := infra.NewAzureResourceManager(azCli)
 
-	operations, err := resourceManager.GetDeploymentResourceOperations(ctx, env.GetSubscriptionId(), env.GetEnvName())
+	resOps, err := resourceManager.GetDeploymentResourceOperations(ctx, env.GetSubscriptionId(), env.GetEnvName())
 	if err != nil {
 		// Status display is best-effort activity.
 		return
 	}
 
+	operations := *resOps
 	succeededCount := 0
+	newlyDeployedResources := []azureutil.DeployedAzureResource{}
```

Comment:
I would use an slice of slices here..  then you don't need `DeployedAzureResource`
like:
```go
newlyDeployedResources := [][]tools.AzCliResourceOperation{}
```

Then, in below loop, you would append an slice directly from operations (main slice) like:

```go
for index, resourceOperation := range operations {
               if resourceOperation.Properties.ProvisioningState == "Succeeded" {
			succeededCount++

			if !loggedResources[resourceOperation.Properties.TargetResource.Id] {
				newlyDeployedResources = append(newlyDeployedResources, operations[index:index+1])
			}
		}
}
```

All the slices inside `newlyDeployedResources` would be of len = 1.  And there would not be any extra memory copy from operations to a new array in memory :)

Reply:
optionally, you could use an slice of pointers to `tools.AzCliResourceOperation`.  like 

```go
newlyDeployedResources := []*tools.AzCliResourceOperation{}
```

then, in the loop, just append the memory address of `resourceOperation` so you can use it later. With no memory copy

Reply:
Yup :) Thanks for the suggestion. This was my original implementation before thinking about batching calls for resolving web app display names

Reply:
Will avoid the memory copy